### PR TITLE
MM-47189:  Migrate "utils/timezone.jsx" to Typescript

### PR DESCRIPTION
**Pull request by AbhinavVihan**
#### Summary
Migrate utils/timezone.jsx to typescript

#### Ticket Link
Fixes https://github.com/mattermost/mattermost-server/issues/21162
JIRA: https://mattermost.atlassian.net/browse/MM-47189

#### Related Pull Requests
None

#### Screenshots
None

#### Release Note

```release-note

```

### CODE REVIEW

File: components/status_dropdown/status_dropdown.tsx [hunk 1]
```diff
@@ -128,7 +128,7 @@ export default class StatusDropdown extends React.PureComponent<Props, State> {
     }
 
     setDnd = (index: number): void => {
-        const currentDate = getCurrentMomentForTimezone(this.props.timezone);
+        const currentDate = getCurrentMomentForTimezone(this.props.timezone as string);
```
Comment (M-ZubairAhmed) [inline components/status_dropdown/status_dropdown.tsx:131]: instead of doing this in so many places we can make the parameter optional

Comment (M-ZubairAhmed) [reply components/status_dropdown/status_dropdown.tsx:131]: and in the function check if its undefined or not then return suitably

File: utils/timezone.tsx [hunk 2]
```diff
@@ -11,20 +11,20 @@ export function getBrowserUtcOffset() {
     return moment().utcOffset();
```
Comment (M-ZubairAhmed) [inline utils/timezone.tsx:11]: also this can be a `ts` file instead of `tsx`

**Review by M-ZubairAhmed — CHANGES_REQUESTED** [review]

File: utils/timezone.tsx (hunk 2 above)
Comment (AbhinavVihan) [reply utils/timezone.tsx:11]: oops, sorry about that.

File: components/status_dropdown/status_dropdown.tsx (hunk 1 above)
Comment (AbhinavVihan) [reply components/status_dropdown/status_dropdown.tsx]: I think the check is already there in the function.

**Review by M-ZubairAhmed — APPROVED** [review]
💯 It looks good to me. Thank you 🎉

**Pull request merged**